src/data/create_segments.py

```diff
@@ -154,46 +160,77 @@ def find_non_ints(roads, int_buffers):
         buffered_lines.append((b, road))
         road_lines_index.insert(idx, b.bounds)
 
-    inter_segments = {'lines': defaultdict(list), 'data': defaultdict(list)}
+    inter_segments = []
     roads_with_int_segments = {}
     count = 0
     print("Generating intersection segments")
+    connected_segment_ids = defaultdict(list)
+
+    # Go through each intersection buffer object
     for i, int_buffer in enumerate(int_buffers):
         util.track(i, 1000, len(int_buffers))
         match_segments = []
         matched_roads = []
-        for idx in road_lines_index.intersection(int_buffer[0].bounds):
+
+        # Add the portion of each road that intersects intersection buffer
+        # to match_segments. These are possible connections
+        # If the road intersects, add that to matched_roads
+        for idx in road_lines_index.intersection(int_buffer.buffer.bounds):
             road = roads[idx]
-            match_segments.append(Segment(road.geometry.intersection(
-                int_buffer[0]), road.properties))
-            matched_roads.append(road)
+            if road.geometry.intersects(int_buffer.buffer):
+                match_segments.append(Segment(road.geometry.intersection(
+                    int_buffer.buffer), road.properties))
+                matched_roads.append(road)
 
-        int_segments = get_connections(int_buffer[1], match_segments)
+        # Get the connections that touch a point in the intersection buffer
+        int_segments = get_connections(int_buffer.points, match_segments)
 
         # Each road_with_int is a road segment and a list of lists of segments
-        # representing the intersections
+        # representing the intersections associated with that road
+        # We store these, because the roads that do not have any intersections
+        # associated with them don't need to be split into separate
+        # intersection and non intersection segments
         # to-do: turn these into intersection objects
         for r in matched_roads:
             if r.properties['id'] not in roads_with_int_segments:
                 roads_with_int_segments[r.properties['id']] = []
             roads_with_int_segments[r.properties['id']] += int_segments
 
         for int_segment in int_segments:
-            inter_segments['lines'][count] = [
-                x.geometry for x in int_segment[0]]
-            inter_segments['data'][count] = [
-                x.properties for x in int_segment[0]]
+
+            # Get the ids of the adjacent non-intersection segments
+            connected = [x.properties['orig_id'] for x in int_segment[0]]
```

Comment:
@j-t-t this int_segment[0] is all the connected segments, right? Orig_id is the id coming from OSM, is that right?

Reply:
Orig_id is set in create_segments: `road.properties['orig_id'] = int(str(99) + str(i))`.

int_segment[0] is a list of connected segments, yes.

I'm using orig_id temporarily for the connected segments because we haven't set the ids at this point in the process. If you look at the end of find_non_ints, you'll see the connected segment ids updates to be the id instead of the orig_id.